### PR TITLE
Set up monorepo

### PR DESCRIPTION
**Pull request by harish-aka-shivi**
Issue Number: #1 #3 
This commit sets up the mono-repo. It also add husky and eslint to the project.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "airbnb",
+  "env": {
+    "jest": true
+  },
+  "rules": {
+    "import/prefer-default-export": 0,
+    "no-restricted-syntax": 0
+}
+}
```
Comment (rishabh1403) [inline .eslintrc]: add new line at the end of file

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (rishabh1403) [inline .gitignore]: same here. please fix all the remaining files

File: backend/app.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+var cookieParser = require('cookie-parser');
+var logger = require('morgan');
+
+var indexRouter = require('./routes/index');
+var usersRouter = require('./routes/users');
+
+var app = express();
+
+// view engine setup
+app.set('views', path.join(__dirname, 'views'));
```
Comment (rishabh1403) [inline backend/app.js]: do you need a view engine?

File: frontend/README.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+
```
Comment (rishabh1403) [inline frontend/README.md]: this readme still valid?

File: frontend/public/index.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (rishabh1403) [inline frontend/public/index.html]: is this the name of your app?

File: frontend/src/serviceWorker.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
+// This optional code is used to register a service worker.
+// register() is not called by default.
```
Comment (rishabh1403) [inline frontend/src/serviceWorker.js]: are you using service workers?

File: backend/app.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
+var createError = require('http-errors');
```
Comment (nimish-gupta) [inline backend/app.js]: use import here.

File: backend/app.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+var express = require('express');
+var path = require('path');
+var cookieParser = require('cookie-parser');
+var logger = require('morgan');
+
+var indexRouter = require('./routes/index');
+var usersRouter = require('./routes/users');
+
+var app = express();
```
Comment (nimish-gupta) [inline backend/app.js]: use `const`

File: backend/bin/www [hunk 9]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+
+/**
+ * Module dependencies.
+ */
+
+var app = require('../app');
```
Comment (nimish-gupta) [inline backend/bin/www]: switch to import.

File: backend/bin/www [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+var debug = require('debug')('backend:server');
+var http = require('http');
+
+/**
+ * Get port from environment and store in Express.
+ */
+
+var port = normalizePort(process.env.PORT || '4000');
```
Comment (nimish-gupta) [inline backend/bin/www]: use `const`

File: frontend/public/index.html (hunk 5 above)
Comment (harish-aka-shivi) [reply frontend/public/index.html]: Resolving.

File: backend/app.js (hunk 3 above)
Comment (harish-aka-shivi) [reply backend/app.js]: I will remove the boilerplate express code.

File: frontend/src/serviceWorker.js (hunk 6 above)
Comment (harish-aka-shivi) [reply frontend/src/serviceWorker.js]: We are not using it now, but we will use the service workers when the app is little stable.

File: backend/app.js (hunk 7 above)
Comment (harish-aka-shivi) [reply backend/app.js]: Sure thing.

File: .gitignore (hunk 2 above)
Comment (harish-aka-shivi) [reply .gitignore]: Sure.

File: frontend/README.md (hunk 4 above)
Comment (harish-aka-shivi) [reply frontend/README.md]: 🤦, Embarrassing mistake, but resolved now.

File: backend/app.js (hunk 8 above)
Comment (harish-aka-shivi) [reply backend/app.js]: Fixed now.

File: backend/bin/www (hunk 9 above)
Comment (harish-aka-shivi) [reply backend/bin/www]: Fixed now.

File: frontend/public/index.html (hunk 5 above)
Comment (harish-aka-shivi) [reply frontend/public/index.html]: Resolved

File: backend/app.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+import cookieParser from 'cookie-parser';
+import logger from 'morgan';
+
+const app = express();
+
+app.use(logger('dev'));
+app.use(json());
+app.use(urlencoded({ extended: false }));
+app.use(cookieParser());
+// app.use(static(join(__dirname, 'public')));
```
Comment (nimish-gupta) [inline backend/app.js]: comments?

File: backend/package.json [hunk 12]
```diff
@@ -25,7 +32,11 @@
   "dependencies": {
     "express": "^4.17.1",
     "npm-run-all": "^4.1.5",
-    "rimraf": "^3.0.0"
+    "rimraf": "^3.0.0",
+    "cookie-parser": "~1.4.4",
+    "debug": "~2.6.9",
+    "http-errors": "~1.6.3",
+    "jade": "~1.11.0",
```
Comment (nimish-gupta) [inline backend/package.json]: jade?

File: frontend/src/App.js [hunk 13]
```diff
@@ -0,0 +1,19 @@
+// eslint-disable-next-line import/no-unresolved
+import React from 'react';
+import logo from './logo.svg';
+import './App.css';
+
+function App() {
```
Comment (nimish-gupta) [inline frontend/src/App.js]: arrow function?

Comment (harish-aka-shivi) [reply frontend/src/App.js]: Fixed this issue.

File: backend/package.json (hunk 12 above)
Comment (harish-aka-shivi) [reply backend/package.json]: I removed these extra dependencies from package.json. 😀

File: backend/app.js (hunk 11 above)
Comment (harish-aka-shivi) [reply backend/app.js]: I changed the file after rebasing, So this commented code is gone.

File: backend/package.json (hunk 12 above)
Comment (Lordtien) [reply backend/package.json]: Did you resolve this? @harish-aka-shivi

**Pull request merged**